### DISCUSSION TITLE
Tuples?

### DISCUSSION THREAD

**Discussion by BioNCphilic** [earthlab-education/Earth-Analytics-AY24#381] [Q&A] [answered]
My code looks a lot like the code in the instructional video, but my result is different.
![image](https://github.com/user-attachments/assets/08110833-1a67-48df-8064-7a9b4dccd910)

Same, except temp vs. precip
![image](https://github.com/user-attachments/assets/8e0739a9-ecc0-4f67-8f4c-7f960112a461)

**Comment by lauren-pease** [comment, marked as answer]
Hi @BioNCphilic, it looks like the Jupyter notebook does not know that annual_climate_df is supposed to be a pandas dataframe. 

First, I would make sure to run the command `%store -r` at the top of the notebook so that you load in previously created variables in the Python environment associated with the notebooks like the annual_climate_df. Otherwise, the environment will interpret the variable as a Python [tuple](https://www.w3schools.com/Python/python_tuples.asp) data type. 

Then, I would check the data type of annual_climate_df by running this command: `type(annual_climate_df)`. type() is a function in Python that allows you to give it a value and return it's data type. For the output you should expect **pandas.core.frame.DataFrame**.

**Reply by BioNCphilic** [reply]
OK - with a bit of restarting, I'm showing my annual_climate_df as a dataframe variable. Thanks @lauren-alexandra
